apps/vscode-extension/src/provider/markdownWebviewProvider.ts

```diff
@@ -0,0 +1,316 @@
+import type { ExtensionContext } from "vscode";
+import * as vscode from "vscode";
+// import { getUserLangs, getUserTheme } from "vscode-shiki-bridge";
+import {
+  createHighlighter,
+  Highlighter,
+  type LanguageRegistration,
+  type RawGrammar,
+} from "shiki";
+
+export function registerMarkdownWebviewProvider(context: ExtensionContext) {
+  const provider = new MarkdownWebviewProvider(context);
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "prettyTsErrors.openMarkdownPreview",
+      async (uri: vscode.Uri) => {
+        if (!uri || !(uri instanceof vscode.Uri)) {
+          console.error(
+            "expected uri to be an instance of vscode.Uri, instead got: ",
+            uri
+          );
+          return;
+        }
+        await provider.openMarkdownPreview(uri);
+      }
+    )
+  );
+}
+
+/**
+ * @see https://github.com/microsoft/vscode-extension-samples/blob/main/webview-sample
+ */
+export class MarkdownWebviewProvider {
+  static readonly viewType = "prettyTsErrors.markdownPreview";
+  private highlighter: Highlighter | null = null;
+  private webviewRootUri: vscode.Uri;
+  private webviewHtmlTemplate: Promise<string>;
+
+  constructor(private readonly context: vscode.ExtensionContext) {
+    this.webviewRootUri = vscode.Uri.joinPath(
+      this.context.extensionUri,
+      "webview"
+    );
+    this.webviewHtmlTemplate = this.loadWebviewHtmlTemplate();
+  }
+
+  private async loadWebviewHtmlTemplate(): Promise<string> {
+    const htmlTemplateUri = vscode.Uri.joinPath(
+      this.webviewRootUri,
+      "index.html"
+    );
+    const htmlTemplateBytes = await vscode.workspace.fs.readFile(
+      htmlTemplateUri
+    );
+    const htmlTemplate = new TextDecoder("utf-8").decode(htmlTemplateBytes);
+    return htmlTemplate;
+  }
+
+  private async loadTypeGrammar(): Promise<RawGrammar | null> {
+    try {
+      const uri = vscode.Uri.joinPath(
+        this.context.extensionUri,
+
+        "syntaxes",
+
+        "type.tmGrammar.json"
+      );
+      const raw = await vscode.workspace.fs.readFile(uri);
+      const json = JSON.parse(new TextDecoder("utf-8").decode(raw));
+      return json as RawGrammar;
+    } catch {
+      console.error("failed to load type grammar");
+      return null;
+    }
+  }
+
+  private async initializeHighlighter() {
+    if (this.highlighter) {
+      return;
+    }
+
+    const customLangs: LanguageRegistration[] = [];
+    const typeGrammar = await this.loadTypeGrammar();
+    if (typeGrammar) {
+      customLangs.push({
+        ...typeGrammar,
+        // Allow using "type" as a language id when calling codeToHtml
+        aliases: ["type"],
+      } as LanguageRegistration);
+    }
+
+    this.highlighter = await createHighlighter({
+      themes: ["dark-plus", "light-plus"],
```

Comment:
Can we use the user's theme here after your changes in the `vscode-shiki-bridge`?